internal-api/src/main/java/datadog/trace/api/ClassloaderConfigurationOverrides.java

```diff
@@ -0,0 +1,112 @@
+package datadog.trace.api;
+
+import datadog.trace.api.config.GeneralConfig;
+import datadog.trace.api.env.CapturedEnvironment;
+import datadog.trace.api.remoteconfig.ServiceNameCollector;
+import datadog.trace.bootstrap.instrumentation.api.AgentSpan;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.WeakHashMap;
+import java.util.function.Function;
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+
+public class ClassloaderConfigurationOverrides {
+  public static final String DATADOG_TAGS_PREFIX = "datadog/tags/";
+  public static final String DATADOG_TAGS_JNDI_PREFIX = "java:comp/env/" + DATADOG_TAGS_PREFIX;
+  // not final for testing purposes
+  static boolean CAN_SPLIT_SERVICE_NAME_BY_DEPLOYMENT =
+      Config.get().isJeeSplitByDeployment() && !Config.get().isServiceNameSetByUser();
+
+  private static class Lazy {
+    private static final ClassloaderConfigurationOverrides INSTANCE =
```

Comment:
```suggestion
 static class Lazy {
    static final ClassloaderConfigurationOverrides INSTANCE =
```
making these package-private means `javac` won't generate a synthetic accessor method (small perf boost)